src/components/GeneralProductCard/GeneralProductCard.tsx

```diff
@@ -38,6 +43,10 @@ export default function GeneralProductCard({
     extraClasses = `${extraClasses} border-b border-black md:border-none`;
   }
 
+  const handleAddToCartClick = useCallback(() => {
+    addProductToCart(product);
+  }, [addProductToCart, product]);
```

Comment:
is there a particular reason for the `useCallback` here? I think in this case it's not necessary

Reply:
I am passing it to another component so I thought it would be necessary to avoid it rerendering unnecessarily , maybe it is not necessary, I'm not really sure 🤔